src/app/components/study-accordion/study-accordion.directive.spec.js

```diff
@@ -29,43 +31,96 @@ describe('studyAccordion', function() {
+            name: '_embedded-name-2',
+            fullName: '_embedded-fullname-2',
+            metadata: '_embedded-metadata-2'
+          }
+        }
+      }
+    ];
+
+    beforeEach(function () {
+      _scope = scope;
+      _scope.prev_node = _nodes[0];
+    });
+
+    it('should recognize it is the same node as the previous one, and the popover selection is empty', function () {
+      var result = _scope.clearMetadata(_nodes[0]);
+      expect(result.isSame).toEqual(true);
+      expect(result.popover.length).toEqual(0);
+    });
+
+    it('should recognize it is a different node from the previous one, and the popover selection is empty', function () {
+      var result = _scope.clearMetadata(_nodes[1]);
+      expect(result.isSame).toEqual(false);
+      expect(result.popover.length).toEqual(0);
+    });
+
+  });
+
   describe('$scope.displayMetadata', function () {
     var _scope, _nodes = [
-      {restObj : {fullName : 'restObj-fullname', metadata : 'restObj-metadata'}, title : 'restObj-title'},
-      {_embedded : {ontologyTerm : {name : '_embedded-name', fullName : '_embedded-fullname', metadata : '_embedded-metadata'}}}
+      {restObj: {fullName: 'restObj-fullname', metadata: 'restObj-metadata'}, title: 'restObj-title'},
+      {
+        _embedded: {
+          ontologyTerm: {
+            name: '_embedded-name',
+            fullName: '_embedded-fullname',
+            metadata: '_embedded-metadata'
+          }
+        }
+      }
     ];
 
     beforeEach(function () {
       _scope = scope;
     });
 
-    it ('should use metadata from restObj when node has restObj property', function () {
-        _scope.displayMetadata(_nodes[0]);
-        expect(_scope.metadataObj.title).toEqual('restObj-title');
-        expect(_scope.metadataObj.fullname).toEqual('restObj-fullname');
-        expect(_scope.metadataObj.body).toEqual('restObj-metadata');
+    it('should use metadata from restObj when node has restObj property', function () {
+      _scope.displayMetadata(_nodes[0]);
+      expect(_scope.metadataObj.title).toEqual('restObj-title');
+      expect(_scope.metadataObj.fullname).toEqual('restObj-fullname');
+      expect(_scope.metadataObj.body).toEqual('restObj-metadata');
     });
 
-    it ('should use metadata from embedded.ontologyTerm when node has _embedded property', function () {
-        _scope.displayMetadata(_nodes[1]);
-        expect(_scope.metadataObj.title).toEqual('_embedded-name');
-        expect(_scope.metadataObj.fullname).toEqual('_embedded-fullname');
-        expect(_scope.metadataObj.body).toEqual('_embedded-metadata');
+    it('should use metadata from embedded.ontologyTerm when node has _embedded property', function () {
+      _scope.displayMetadata(_nodes[1]);
+      expect(_scope.metadataObj.title).toEqual('_embedded-name');
+      expect(_scope.metadataObj.fullname).toEqual('_embedded-fullname');
+      expect(_scope.metadataObj.body).toEqual('_embedded-metadata');
     });
 
```

Comment:
Maybe more specification to : 
- Make sure that $scope.clearMetadata is called when displayMetadata invoked
- specification when node is not existing